includes/admin/extensions/abstract-extension.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+namespace EDD\Admin\Extensions;
+
+use \EDD\Admin\Pass_Manager;
+
+abstract class Extension {
+
+	/**
+	 * The product ID. This only needs to be set if the extending class is
+	 * for a single product.
+	 *
+	 * @since 2.11.x
+	 * @var int
+	 */
+	protected $item_id;
+
+	/**
+	 * The required AA pass level.
+	 */
+	const PASS_LEVEL = Pass_Manager::PERSONAL_PASS_ID;
+
+	/**
+	 * The Extension Manager
+	 *
+	 * @var \EDD\Admin\Extensions\Extension_Manager
+	 */
+	protected $manager;
+
+	public function __construct() {
+		$this->manager = new \EDD\Admin\Extensions\Extension_Manager( static::PASS_LEVEL );
+	}
+
+	/**
+	 * Whether the extension is activated.
+	 *
+	 * @since 2.11.x
+	 * @return bool
+	 */
+	abstract protected function is_activated();
+
+	/**
+	 * Output the settings field (installation helper).
+	 *
+	 * @param array $args
+	 * @return void
+	 */
+	public function settings_field() {
+		if ( $this->is_activated() ) {
+			return;
+		}
+		$this->do_single_extension_card();
+	}
+
+	/**
+	 * Outputs a single extension card.
+	 *
+	 * @since 2.11.x
+	 * @param bool|int $item_id Optional: the individual extension product ID.
+	 * @return void
+	 */
+	public function do_single_extension_card( $item_id = false ) {
+		if ( empty( $item_id ) && empty( $this->item_id ) ) {
+			return;
+		}
+		$product_data = $this->get_product_data( $item_id );
+		if ( ! $product_data || empty( $product_data['title'] ) ) {
+			return;
+		}
+		$this->manager->do_extension_card(
+			$product_data,
+			$this->get_button_parameters( $product_data, $item_id ),
+			$this->get_link_parameters( $product_data )
+		);
+	}
+
+	/**
+	 * Gets the product data for a specific extension.
+	 *
+	 * @param boolean $item_id
+	 * @return bool|object False if there is no data; API object if there is.
```

Comment:
```suggestion
	 * @return false|array False if there is no data; API object if there is.
```

Looks like it's actually an array, not an object.